samples/core/tutorials/estimators/linear.ipynb

```diff
@@ -1399,6 +1422,211 @@
+      "cell_type": "markdown",
+      "source": [
+        "## Adding Regularization to Prevent Overfitting\n",
+        "\n",
+        "Regularization is a technique used to avoid **overfitting**. Overfitting happens\n",
+        "when your model does well on the data it is trained on, but worse on test data\n",
+        "that the model has not seen before, such as live traffic. Overfitting generally\n",
+        "occurs when a model is excessively complex, such as having too many parameters\n",
+        "relative to the number of observed training data. Regularization allows for you\n",
+        "to control your model's complexity and makes the model more generalizable to\n",
+        "unseen data.\n",
+        "\n",
+        "You can add L1 and L2 regularizations to the model with the following code:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "lzMUSBQ03hHx",
+        "colab_type": "code",
+        "colab": {
+          "autoexec": {
+            "startup": false,
+            "wait_interval": 0
+          }
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "model_l1 = tf.estimator.LinearClassifier(\n",
+        "    feature_columns=base_columns + crossed_columns,\n",
+        "    optimizer=tf.train.FtrlOptimizer(\n",
+        "        learning_rate=0.1,\n",
+        "        l1_regularization_strength=10.0,\n",
+        "        l2_regularization_strength=0.0))\n",
+        "\n",
+        "model_l1.train(train_inpf)\n",
+        "\n",
+        "results = model_l1.evaluate(test_inpf)\n",
+        "clear_output()\n",
+        "for key in sorted(results):\n",
+        "  print('%s: %0.2f' % (key, results[key]))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "ofmPL212JIy2",
+        "colab_type": "code",
+        "colab": {
+          "autoexec": {
+            "startup": false,
+            "wait_interval": 0
+          }
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "model_l2 = tf.estimator.LinearClassifier(\n",
+        "    feature_columns=base_columns + crossed_columns,\n",
+        "    optimizer=tf.train.FtrlOptimizer(\n",
+        "        learning_rate=0.1,\n",
+        "        l1_regularization_strength=0.0,\n",
+        "        l2_regularization_strength=10.0))\n",
+        "\n",
+        "model_l2.train(train_inpf)\n",
+        "\n",
+        "results = model_l2.evaluate(test_inpf)\n",
+        "clear_output()\n",
+        "for key in sorted(results):\n",
+        "  print('%s: %0.2f' % (key, results[key]))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Lp1Rfy_k4e7w",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "These regularized models don't perform very differently base model. Let look ar the models' weight distributions to better see the effect of the regularization:"
```

Comment:
Copy/typo edit:

These regularized models don't perform much better than the base model. Let's look at the model's weight distributions to better see the effect of the regularization:

Reply:
Done